src/sentry/notifications/notifications/base.py

```diff
@@ -86,8 +88,15 @@ def get_type(self) -> str:
     def get_unsubscribe_key(self) -> tuple[str, int, str | None] | None:
         return None
 
-    def record_notification_sent(self, recipient: Team | User, provider: ExternalProviders) -> None:
-        raise NotImplementedError
+    def record_notification_sent(
+        self, recipient: Team | User, provider: ExternalProviders, **kwargs: Any
+    ) -> None:
+        analytics.record(
+            f"integrations.{provider.name}.notification_sent",
```

Comment:
@mgaeta You should add these events to ETL so we can see them in Amplitude: https://github.com/getsentry/etl/blob/master/etl/operators/analytics_events_schema.py